### PR TITLE
Tagging of created network resources

### LINKED ISSUE #171: Tag the OpenStack resources created by CAPO
#150 added network, subnet, and router creation as part of the cluster reconciler. We'll likely be adding logic to create other resources like SecurityGroups (#170). We should tag all these resources in order to ease their management not only for the provider itself but also for operators.

Support for Neutron tags was recently added: https://github.com/gophercloud/gophercloud/pull/1259

### PR DESCRIPTION
**Pull request by chrigl**
**What this PR does / why we need it**:

Tagged resouces networks, subnet and router with
`cluster-api-provider-openstack` and the cluster name in form of
`<namespace>/<name>`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #171

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/openstack/clients/networkservice.go [hunk 1]
```diff
@@ -115,13 +116,22 @@ func (s *NetworkService) reconcileNetwork(networkName string, desired openstackc
 		return emptyNetwork, err
 	}
 
+	_, err = attributestags.ReplaceAll(s.client, "networks", network.ID, attributestags.ReplaceAllOpts{
+		Tags: []string{
+			"cluster-api-provider-openstack",
+			clusterName,
+		}}).Extract()
+	if err != nil {
+		return emptyNetwork, err
```
Comment (flaper87) [inline pkg/cloud/openstack/clients/networkservice.go]: I wonder if we should return the error or just ignore it. By returning it we're requiring OpenStack deployments to have this endpoint available (we'll be requiring newer versions of OpenStack).

FWIW, I prefer this but I wanted to make sure we're on the same page here.

**Comment by chaosaffe** [comment]
Thanks @flaper87

**Pull request merged**